test/built-ins/Promise/prototype/finally/rejected-observable-then-calls.js

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2017 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+author: Jordan Harband
+description: finally observably calls .then
+esid: sec-promise.prototype.finally
+features: [Promise.prototype.finally]
+flags: [async]
+---*/
+
+var initialThenCount = 0;
+var noReason = {};
+var no = Promise.reject(noReason);
+no.then = function () {
+  initialThenCount += 1;
+  return Promise.prototype.then.apply(this, arguments);
+};
+
+var onFinallyThenCount = 0;
+var yesValue = {};
+var yes = Promise.resolve(yesValue);
+yes.then = function () {
+  onFinallyThenCount += 1;
+  return Promise.prototype.then.apply(this, arguments);
+};
+
+var finallyCalled = false;
+var catchCalled = false;
+
+no.catch(function (e) {
+  assert.sameValue(e, noReason);
+  throw e;
```

Comment:
It seems  that we need test that we raise TypeError when we use `finally` with `new` i.e.
```
const p = new Promise((resolve)=>{});

assert.throws(TypeError, function() {
  const f = new p.finally();
}, 'no constructor');
```